service/src/main/java/com/epam/esm/certificate/CertificateService.java

```diff
@@ -0,0 +1,150 @@
+package com.epam.esm.certificate;
+
+import com.epam.esm.exception.ServiceConflictException;
+import com.epam.esm.tag.Tag;
+import com.epam.esm.tag.TagDao;
+import com.epam.esm.tag.TagDto;
+import lombok.extern.slf4j.Slf4j;
+import org.modelmapper.ModelMapper;
+import org.springframework.hateoas.PagedModel;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.ConstraintViolationException;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import java.math.BigDecimal;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+@Transactional
+public class CertificateService {
+    private final CertificateDao certificateDao;
+    private final TagDao tagDao;
+    private final ModelMapper modelMapper;
+
+    public CertificateService(TagDao tagDao,
+                              CertificateDao certificateDao,
+                              ModelMapper modelMapper) {
+        this.tagDao = tagDao;
+        this.certificateDao = certificateDao;
+        this.modelMapper = modelMapper;
+    }
+
+    public CertificateDto create(CertificateDto certificateDto) {
+        Certificate certificate = modelMapper.map(certificateDto, Certificate.class);
+        certificateDao.create(certificate);
+        return modelMapper.map(certificate, CertificateDto.class);
+    }
+
+    public void update(CertificateDto certificateDto) {
+        Certificate certificate = modelMapper.map(certificateDto, Certificate.class);
+        long certificateId = certificate.getId();
+        Certificate beforeUpdate = certificateDao.find(certificateId)
+                .orElseThrow(() -> new CertificateNotFoundException("Certificate with id "
+                        + certificate.getId() + " doesn't exist"));
+        certificate.setCreationDate(beforeUpdate.getCreationDate());
+        certificateDao.update(certificate);
+    }
+
+    @Transactional
```

Comment:
do we need transnational method here? the class itself is already marked as transactional